crates/gitbutler-branch-actions/src/virtual.rs

```diff
@@ -455,7 +456,8 @@ pub fn list_virtual_branches_cached(
         let base_current = true;
 
         let upstream = upstream_branch.and_then(|upstream_branch| {
-            let remotes = repo.remotes().ok()?;
+            let remotes_raw = repo.remotes().ok()?;
+            let remotes = remotes_raw.into_iter().flatten().collect_vec();
```

Comment:
@Byron do you have an opinion on itertools? In recent weeks I've seen it get used more and more, and none of the uses seem to be much of a gain over doing it without.

Reply:
The alternative feels a bit more verbose? Either:
```
let remotes: Vec<_> = remotes_raw.into_iter().flatten().collect();
```
or
```
let remotes = remotes_raw.into_iter().flatten().collect::<Vec<_>>();
```

Reply:
I think if they are functionally I don't mind using `.collect()`, but which of the two styles I mention above do you guys prefer?

Reply:
collect_vec() is the itertools, so the plain rust collect() is the preferable one.

Reply:
Yes but it often requires explicit types. Just wondering if there's a better option between putting it at the beginning of the line, or next to the collect_vec call, as mentioned in my previous comment.

Reply:
I think `let remotes = remotes_raw.into_iter().flatten().collect::<Vec<_>>();` is preferable

Reply:
Sometimes the `Vec<_>` can be inferred, but probably not in this situation.

To my mind, using `itertools` here is fine as it is intuitive and thus is fast and easy to pull off. And that matters.
For `gitoxide` I can't use it, as I get angry letters sent to my house when a new dependency shows up downstream, and it could hide allocations which I rather keep explicit (e.g. `it.sorted()`).

But in application code I don't think there is a good enough reason not to use it.